ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/sse/ClientSSESession.kt

```diff
@@ -19,9 +31,157 @@ public interface SSESession : CoroutineScope {
     public val incoming: Flow<ServerSentEvent>
 }
 
+/**
+ * A Server-sent events session.
+ *
+ * Example of usage:
+ * ```kotlin
+ * client.sse({
+ *     url("http://localhost:8080/serverSentEvents")
+ * }, deserialize = {
+ *     typeInfo, jsonString ->
+ *     val serializer = Json.serializersModule.serializer(typeInfo.kotlinType!!)
+ *     Json.decodeFromString(serializer, jsonString)!!
+ * }) {
+ *     incoming.collect { event: TypedServerSentEvent<String> ->
+ *         when (event.event) {
+ *             "customer" -> {
+ *                 val customer: Customer? = deserialize<Customer>(event.data)
+ *             }
+ *             "product" -> {
+ *                 val product: Product? = deserialize<Product>(event.data)
+ *             }
+ *         }
+ *     }
+ * }
+ * ```
+ */
+public interface SSESessionWithDeserialization : CoroutineScope {
+    /**
+     * An incoming server-sent events flow.
```

Comment:
Perhaps we can add more information or a link to check what fields an `event` contains?